### PR TITLE
Update index.yml

### PR DESCRIPTION
**Pull request by lindalu-MSFT**
add graph rest API docs to overview list

### CODE REVIEW

**Comment by lindalu-MSFT** [comment]
Hey @AlexJerabek Like the other landing pages, I updated the links to Q&A and Ideas forum. But I also wanted to add the Graph OneNote REST API docs to this landing page. I fiddled around in a previous PR, but since there essentially TWO docsets on this landing, I wanted to clear this with you.

**Comment by AlexJerabek** [comment]
@lindalu-MSFT Can you please coordinate with @Rick-Kirkham about linking to the REST APIs from here. I'm not sure how that content is related.

File: docs/onenote/index.yml [hunk 1]
```diff
@@ -23,8 +23,10 @@ landingContent:
         links:
           - text: What are OneNote add-ins?
             url: onenote-add-ins-programming-overview.md
-          - text: OneNote API Overview
-            url: /graph/integrate-with-onenote
+          - text: OneNote JavaScript API Overview
+            url: ../reference/overview/onenote-add-ins-javascript-reference.md         
+          - text: Microsoft Graph OneNote REST API Overview
+            url: /graph/integrate-with-onenote.md
```
Comment (ElizabethSamuel-MSFT) [inline docs/onenote/index.yml:29]:
```suggestion
            url: /graph/integrate-with-onenote
```

**Comment by lindalu-MSFT** [comment]
@Rick-Kirkham We are deprecating the [OneNote docshub page](https://developer.microsoft.com/en-us/onenote/docs) on developer.microsoft.com and forwarding to this landing page. Note that there is a link to the Graph OneNote REST API. My thought was that since this page is the new landing page, we should link to the Graph OneNote REST API here as well.

**Comment by ElizabethSamuel-MSFT** [comment]
Confirmed with @Rick-Kirkham offline that this change is good to go.

**Comment by lindalu-MSFT** [comment]
Thanks for straightening this out!

**Pull request merged**